src/RequestHandle.cpp

```diff
@@ -587,14 +610,24 @@ auto curl_write_partial_data(
     // Keep track of total number of bytes received
     raw_request_ptr->m_total_bytes_received += data_length;
 
+    size_t bytes_left_to_write;
+
+    if (raw_request_ptr->m_bytes_left_to_write > -1)
+    {
+        bytes_left_to_write = static_cast<size_t>(raw_request_ptr->m_bytes_left_to_write);
+    }
+    else
+    {
+        bytes_left_to_write = 0;
```

Comment:
just initialize the variable to 0 and then you can remove the else block

Reply:
Done!